techniques/fileConfiguration/fileManagement/4.1/manageFilesAndFolders.st

```diff
@@ -218,85 +236,93 @@ bundle agent manageFilesAndFolders_RudderUniqueID {
+        ifvarclass => "manageSyms_${index}_RudderUniqueID_kept.!manageSyms_${index}_RudderUniqueID_repaired";
 
       "any" usebundle => rudder_common_report_index("manageFilesAndFolders", "result_repaired", "${file[${index}][uuid]}", "File", "${file[${index}][path]}", "The symlink ${file[${index}][path]} was successfully created", "${index}"),
-				ifvarclass => "manageSyms_${index}_RudderUniqueID_repaired";
+        ifvarclass => "manageSyms_${index}_RudderUniqueID_repaired";
 
       "any" usebundle => rudder_common_report_index("manageFilesAndFolders", "result_error", "${file[${index}][uuid]}", "File", "${file[${index}][path]}", "The symlink ${file[${index}][path]} could not be created", "${index}"),
-				ifvarclass => "manageSyms_${index}_RudderUniqueID_error";
+        ifvarclass => "manageSyms_${index}_RudderUniqueID_error";
 
-			## Reporting for no change
+      ## Reporting for no change
       "any" usebundle => rudder_common_report_index("manageFilesAndFolders", "result_success", "${file[${index}][uuid]}", "File", "${file[${index}][path]}", "No file creation action specified for ${file[${index}][path]}", "${index}"),
                                 ifvarclass => "manageFilesAndFolders_nocreate_${index}";
-			
-			# Reporting for the permissions
-			## NA if not set
+      
+      # Reporting for the permissions
+      ## NA if not set
       "any" usebundle => rudder_common_report_index("manageFilesAndFolders", "result_na", "${file[${index}][uuid]}", "Permissions", "${file[${index}][path]}", "The item ${file[${index}][path]} was not requested to have its permission enforced", "${index}"),
-				ifvarclass => "!manageFilesAndFolders_check_permissions_${index}";
+        ifvarclass => "!manageFilesAndFolders_check_permissions_${index}";
+
+      ##NA if the path does not exist, no creation rules setted and permissions check set to true
+      "any" usebundle => rudder_common_report_index("manageFilesAndFolders", "result_na", "${file[${index}][uuid]}", "Permissions", "${file[${index}][path]}", "The item ${file[${index}][path]} was not found but not requested to be created", "${index}"),
+        ifvarclass => "manageFilesAndFolders_check_permissions_${index}.!manageFilesAndFolders_exists_${index}.manageFilesAndFolders_nocreate_${index}";
+
+      ## Error if we are not in NA and there is no class defined or NA conditions verified
+      "any" usebundle => rudder_common_report_index("manageFilesAndFolders", "result_error", "${file[${index}][uuid]}", "Permissions", "${file[${index}][path]}", "The item ${file[${index}][path]} was not found", "${index}"),
+        ifvarclass => "manageFilesAndFolders_check_permissions_${index}.!manageFilesAndFolders_perms_${index}_RudderUniqueID_ok.!manageFilesAndFolders_perms_${index}_RudderUniqueID_not_ok.!(manageFilesAndFolders_check_permissions_${index}.!manageFilesAndFolders_exists_${index}.manageFilesAndFolders_nocreate_${index})";
```

Comment:
can you explain more this condition, and how it is supposed to work ? I had to look back at the integrality of code to see what it was doing